Services/src/CSClassroom/CSClassroom.Service/Projects/SubmissionService.cs

```diff
@@ -687,12 +738,34 @@ await _emailProvider.SendMessageAsync
 		}
 
 		/// <summary>
-		/// Returns a query for all submissions in a given checkpoint.
+		/// Returns a query for all submissions in a given checkpoint
+		/// for a single section, or for all sections if section == null.
 		/// </summary>
 		private IQueryable<Submission> GetCheckpointSubmissionsQuery(
 			Checkpoint checkpoint,
 			Section section)
 		{
+			// Identical queries, except leave off the SectionId filter if section == null
```

Comment:
Done!  But now you must solve this c# riddle.  Because you are Midas, your approach works beautifully:

```
	sm => (section == null || sm.SectionId == section.Id)
		  && sm.Role == SectionRole.Student
```

Originally, I tried something similar, as I too didn't want to duplicate that query code.  However, since I am Typhoid Mary, this is what my magic touch wrought:

```
	sm => (section == null) ? sm.Role == SectionRole.Student  :
		(sm.SectionId == section.Id && sm.Role == SectionRole.Student)
```

SHOULD be equivalent, but it resulted in a System.NullReferenceException:

`System.InvalidOperationException: An exception was thrown while attempting to evaluate a LINQ query parameter expression. To show additional information call EnableSensitiveDataLogging() when overriding DbContext.OnConfiguring. ---> System.NullReferenceException: Object reference not set to an instance of an object.`

I surmised that LINQ / EF were doing some reflection & manual evaluation of my lambda in order to convert it into sql, and wasn't smart enough to deal with expressions that might contain nulls, even if the C# would handle them properly.  So I had chosen to stop being clever and just duped the query code to avoid tricking LINQ / EF.  But now that I see your approach works, I'm using that instead.

So... why would short-circuiting work, but a conditional expression fail?

Reply:
Very likely for the reason you said (EF does not support all possible expression trees).